### PR TITLE
close http.Response Body

### PR DESCRIPTION
**Pull request by bboozzoo**
make sure that we close Body of each http.Response

@kacf @pasinskim @GregorioDiStefano

### CODE REVIEW

File: client_update.go [hunk 1]
```diff
@@ -86,8 +86,10 @@ func (u *UpdateClient) FetchUpdate(api ApiRequester, url string) (io.ReadCloser,
 	}
 
 	if r.ContentLength < 0 {
+		defer r.Body.Close()
```
Comment (pasinskim) [inline client_update.go]: What about closing this couple of lines earlier - here: https://github.com/mendersoftware/mender/pull/95/files#diff-c1a9cef21663a7f28d8d2eb327d47bffR80

Comment (bboozzoo) [inline client_update.go]: Uhh right, we should also close it there.

Comment (bboozzoo) [inline client_update.go]: fixed

Comment (pasinskim) [inline client_update.go]: Why we can not have just one `defer r.Body.Close()` after (or before) `log.Debugf("Received fetch update response %v+", r)`?

Comment (bboozzoo) [inline client_update.go]: Because we still want to receive the update ;) otherwise io stream will get closed when fetch update returns.

Comment (pasinskim) [inline client_update.go]: Crap. You are right. I did not follow the logic. In this case I think you can skip `defer` and close the stream explicitly. What is more, you are right that we need to close the stream in `func (u *UpdateInstallState) Handle(ctx *StateContext, c Controller) (State, bool)`.
Can you add something like below:

```
func (u *UpdateInstallState) Handle(ctx *StateContext, c Controller) (State, bool) {

	defer u.imagein.Close()

	// start deployment logging
	DeploymentLogger.Enable(u.update.ID)

```

Comment (bboozzoo) [inline client_update.go]: Right, fixed!

**Comment by pasinskim** [comment]
Looks good now.

**Pull request merged**